src/ServicePulse.Host/vue/src/components/monitoring/EndpointDetails.vue

```diff
@@ -61,6 +62,18 @@ watch(monitoringStore.historyPeriod, (newValue) => {
   changeRefreshInterval(newValue.refreshIntervalVal);
 });
 
+const paginatedMessageTypes = computed(() => {
+  return endpoint.value.messageTypes.slice((endpoint.value.messageTypesPage - 1) * endpoint.value.messageTypesItemsPerPage, endpoint.value.messageTypesPage * endpoint.value.messageTypesItemsPerPage);
+});
+
+watch(
+  () => endpoint.value.messageTypesPage,
+  () => {
+    const breakdownTabName = showInstancesBreakdown ? "instancesBreakdown" : "messageTypeBreakdown";
+    router.push({ name: "endpoint-details", params: { endpointName: endpointName }, query: { historyPeriod: historyPeriod.value.pVal, tab: breakdownTabName, pageNo: endpoint.value.messageTypesPage } });
```

Comment:
as per the cleanup branch, I'm not convinced we want every change of tab in browser history

Reply:
Fixed this but the routing still does not behave with the history as expected. I may need to pair up with someone to figure this out, or it may be better to fix it after splitting other components out (to minimize amount of code in the page that is touching it).

Reply:
Agreed, let's try to fix it after splitting other components out.